api/python/tests/integration/test_packages.py

```diff
@@ -623,3 +623,28 @@ def test_manifest():
 
     pkg2 = Package.browse(pkg_hash=top_hash)
     assert list(pkg.manifest) == list(pkg2.manifest)
+
+def test_map():
+    pkg = Package()
+    pkg.set('as/df', LOCAL_MANIFEST)
+    pkg.set('as/qw', LOCAL_MANIFEST)
+    assert set(pkg.map(lambda lk, entry: lk)) == {'as/df', 'as/qw'}
+
+
+def test_filter():
+    pkg = Package()
+    pkg.set('as/df', LOCAL_MANIFEST)
+    pkg.set('as/qw', LOCAL_MANIFEST)
+    assert pkg.filter(lambda lk, entry: lk == 'as/df') == [('as/df', pkg['as/df'])]
+
+
+def test_reduce():
+    pkg = Package()
+    pkg.set('as/df', LOCAL_MANIFEST)
+    pkg.set('as/qw', LOCAL_MANIFEST)
+    assert pkg.reduce(lambda a, b: a) == ('as/df', pkg['as/df'])
+    assert pkg.reduce(lambda a, b: b) == ('as/qw', pkg['as/qw'])
```

Comment:
This seems like a werid implementation. I would expect `lambda a, b: a` in a reduce to return the first element only of a list, not a tuple.

Reply:
Entries are not aware of their logical key, but the logical key is very useful for any operation you will perform on an entry. So the options were:

* Omit the logical key: `(a, b) = left_entry, right_entry`
* Include the logical key and make the reducer operate on tuples: `(a, b) = (left_lk, left_entry), (right_lk, right_entry)`
* Include the logical keys as separate arguments: `(a_1, a_2, b_1, b_2) = (left_lk, left_entry, right_lk, right_entry)`.

And I chose the second option. What do you think?